sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/HighlightImporter.java

```diff
@@ -29,10 +29,14 @@
 import org.sonar.api.batch.sensor.highlighting.TypeOfText;
 import org.sonarsource.dotnet.protobuf.SonarAnalyzer;
 import org.sonarsource.dotnet.protobuf.SonarAnalyzer.TokenTypeInfo;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
 
 class HighlightImporter extends ProtobufImporter<SonarAnalyzer.TokenTypeInfo> {
 
   private final SensorContext context;
+  private final Map<InputFile, HashSet<TokenTypeInfo.TokenInfo>> fileHighlights = new HashMap<>();
```

Comment:
This should be a List instead of a Set since we don't want to rely on TokenInfo's hashCode/equals.

Reply:
@dbmeneses is this because it is slow, or some other reason? As far as I can see from the implementations, the hashcode() and equals() method will work well, not sure how changing to List will help (as I will have to use equals() anyway)... Can you elaborate more?

Reply:
If the intention is to not keep duplicates in the collection, using a set is correct and we need to make sure that hashcode() and equals() behave as we want (are consistent with what we consider being duplications). Otherwise, the set will use hashcode() and equals() to calculate the hash of the object and detect duplicates.